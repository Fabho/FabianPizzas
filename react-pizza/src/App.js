import React from "react";
import axios from "axios";
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Link
} from "react-router-dom";

export default function App() {
    return (
        <Router>
            <div>
                <nav>
                    <ul>
                        <li>
                            <Link to="/">Home</Link>
                        </li>
                        <li>
                            <Link to="/about">About</Link>
                        </li>
                        <li>
                            <Link to="/users">Users</Link>
                        </li>
                    </ul>
                </nav>

                {/* A <Switch> looks through its children <Route>s and
            renders the first one that matches the current URL. */}
                <Switch>
                    <Route path="/about">
                        <About />
                    </Route>
                    <Route path="/users">
                        <Users />
                    </Route>
                    <Route path="/">
                        <Home />
                    </Route>
                </Switch>
            </div>
        </Router>
    );
}

class Home extends React.Component{
    componentDidMount() {
        axios.get('http://localhost:63764/api/Toppings')
            .then(function (response) {
                // handle success
                console.log("bien")
                console.log(response);
            })
            .catch(function (error) {
                // handle error
                console.log("mal")
                console.log(error)
            });
    }

    render(){
        
            return <h2>Homea</h2>;
        
    }
} 

function About() {
    return <h2>About</h2>;
}

function Users() {
    return <h2>Users</h2>;
}