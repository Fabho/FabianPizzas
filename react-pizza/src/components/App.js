import React from "react";
import axios from "axios";
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Link
} from "react-router-dom";
import Pizzas from './Pizzas';
import Toppings from './Toppings';
import Home from './Home';

export default function App() {
    return (
        <Router>
            <ul className="nav">
                <li className="nav-item">
                    <Link className="nav-link active" to="/">Home</Link>
                </li>
                <li className="nav-item">
                    <Link className="nav-link" to="/pizzas">Pizzas</Link>
                </li>
                <li className="nav-item">
                    <Link className="nav-link" to="/toppings">Toppings</Link>
                </li>
            </ul>
            <div>
                <Switch>
                    <Route exact path="/" component={Home}>
                        <Home />
                    </Route>
                    <Route path="/pizzas" component={Pizzas}>
                        <Pizzas />
                    </Route>
                    <Route path="/toppings" component={Toppings}>
                        <Toppings />
                    </Route>
                </Switch>
            </div>
        </Router>
    );
}