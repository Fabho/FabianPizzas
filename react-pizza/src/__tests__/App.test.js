import React from 'react';
import App from '../components/App';
import { create } from 'react-test-renderer'

describe('Pizzas app Test',()=>{
    test('should render app component', () => {
    let tree = create(<App />)
    expect(tree.toJSON()).toMatchSnapshot();
  })
})