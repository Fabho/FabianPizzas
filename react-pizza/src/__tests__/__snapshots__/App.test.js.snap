// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My first snapshot test should render app component 1`] = `
Array [
  <ul
    className="nav"
  >
    <li
      className="nav-item"
    >
      <a
        className="nav-link active"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/pizzas"
        onClick={[Function]}
      >
        Pizzas
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/toppings"
        onClick={[Function]}
      >
        Toppings
      </a>
    </li>
  </ul>,
  <div>
    <div
      className="container"
    >
      <section
        className="page-section bg-dark"
      >
        <div
          className="container"
        >
          <div
            className="row justify-content-center"
          >
            <div
              className="col-lg-8 text-center"
            >
              <h2
                className="text-white mt-0"
              >
                Pizzas demo App by Fabian Calsina
              </h2>
              <hr
                className="divider light my-4"
              />
              <p
                className="text-white-50 mb-4"
              >
                <strong>
                  Back end:
                </strong>
                .Net webapi2, Entity Framework usingMSSQLLocalDB
              </p>
              <p
                className="text-white-50 mb-4"
              >
                <strong>
                  Front end:
                </strong>
                React, redux, redux-thunk, bootstrap 4.0
              </p>
            </div>
          </div>
        </div>
      </section>
      <section
        className="page-section"
      >
        <div
          className="container"
        >
          <h2
            className="text-center mt-0"
          >
            At Your Service
          </h2>
          <div
            className="text-center"
          >
            <img
              src="fgg.JPG"
              width="300"
            />
          </div>
          <hr
            className="divider my-4"
          />
          <div
            className="row"
          >
            <div
              className="col-lg-3 col-md-6 text-center"
            >
              <div
                className="mt-5"
              >
                <i
                  className="fas fa-4x fa-gem text-primary mb-4"
                />
                <h3
                  className="h4 mb-2"
                >
                  Email
                </h3>
                <p
                  className="text-muted mb-0"
                >
                  fabian.calsina@gmail.com
                </p>
              </div>
            </div>
            <div
              className="col-lg-3 col-md-6 text-center"
            >
              <div
                className="mt-5"
              >
                <i
                  className="fas fa-4x fa-laptop-code text-primary mb-4"
                />
                <h3
                  className="h4 mb-2"
                >
                  Cellphone
                </h3>
                <p
                  className="text-muted mb-0"
                >
                  591 70534341
                </p>
              </div>
            </div>
            <div
              className="col-lg-3 col-md-6 text-center"
            >
              <div
                className="mt-5"
              >
                <i
                  className="fas fa-4x fa-globe text-primary mb-4"
                />
                <h3
                  className="h4 mb-2"
                >
                  Github Profile
                </h3>
                <a
                  href="https://github.com/Fabho"
                  target="_blank"
                >
                  Fabho profile
                </a>
              </div>
            </div>
            <div
              className="col-lg-3 col-md-6 text-center"
            >
              <div
                className="mt-5"
              >
                <i
                  className="fas fa-4x fa-heart text-primary mb-4"
                />
                <h3
                  className="h4 mb-2"
                >
                  Made with Love
                </h3>
                <p
                  className="text-muted mb-0"
                >
                  2020
                </p>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>,
]
`;

exports[`My first snapshot test testing app button 1`] = `
Array [
  <ul
    className="nav"
  >
    <li
      className="nav-item"
    >
      <a
        className="nav-link active"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/pizzas"
        onClick={[Function]}
      >
        Pizzas
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/toppings"
        onClick={[Function]}
      >
        Toppings
      </a>
    </li>
  </ul>,
  <div>
    <div
      className="container"
    >
      <section
        className="page-section bg-dark"
      >
        <div
          className="container"
        >
          <div
            className="row justify-content-center"
          >
            <div
              className="col-lg-8 text-center"
            >
              <h2
                className="text-white mt-0"
              >
                Pizzas demo App by Fabian Calsina
              </h2>
              <hr
                className="divider light my-4"
              />
              <p
                className="text-white-50 mb-4"
              >
                <strong>
                  Back end:
                </strong>
                .Net webapi2, Entity Framework usingMSSQLLocalDB
              </p>
              <p
                className="text-white-50 mb-4"
              >
                <strong>
                  Front end:
                </strong>
                React, redux, redux-thunk, bootstrap 4.0
              </p>
            </div>
          </div>
        </div>
      </section>
      <section
        className="page-section"
      >
        <div
          className="container"
        >
          <h2
            className="text-center mt-0"
          >
            At Your Service
          </h2>
          <div
            className="text-center"
          >
            <img
              src="fgg.JPG"
              width="300"
            />
          </div>
          <hr
            className="divider my-4"
          />
          <div
            className="row"
          >
            <div
              className="col-lg-3 col-md-6 text-center"
            >
              <div
                className="mt-5"
              >
                <i
                  className="fas fa-4x fa-gem text-primary mb-4"
                />
                <h3
                  className="h4 mb-2"
                >
                  Email
                </h3>
                <p
                  className="text-muted mb-0"
                >
                  fabian.calsina@gmail.com
                </p>
              </div>
            </div>
            <div
              className="col-lg-3 col-md-6 text-center"
            >
              <div
                className="mt-5"
              >
                <i
                  className="fas fa-4x fa-laptop-code text-primary mb-4"
                />
                <h3
                  className="h4 mb-2"
                >
                  Cellphone
                </h3>
                <p
                  className="text-muted mb-0"
                >
                  591 70534341
                </p>
              </div>
            </div>
            <div
              className="col-lg-3 col-md-6 text-center"
            >
              <div
                className="mt-5"
              >
                <i
                  className="fas fa-4x fa-globe text-primary mb-4"
                />
                <h3
                  className="h4 mb-2"
                >
                  Github Profile
                </h3>
                <a
                  href="https://github.com/Fabho"
                  target="_blank"
                >
                  Fabho profile
                </a>
              </div>
            </div>
            <div
              className="col-lg-3 col-md-6 text-center"
            >
              <div
                className="mt-5"
              >
                <i
                  className="fas fa-4x fa-heart text-primary mb-4"
                />
                <h3
                  className="h4 mb-2"
                >
                  Made with Love
                </h3>
                <p
                  className="text-muted mb-0"
                >
                  2020
                </p>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>,
]
`;

exports[`Pizzas app Test should render app component 1`] = `
Array [
  <ul
    className="nav"
  >
    <li
      className="nav-item"
    >
      <a
        className="nav-link active"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/pizzas"
        onClick={[Function]}
      >
        Pizzas
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/toppings"
        onClick={[Function]}
      >
        Toppings
      </a>
    </li>
  </ul>,
  <div>
    <div
      className="container"
    >
      <section
        className="page-section bg-dark"
      >
        <div
          className="container"
        >
          <div
            className="row justify-content-center"
          >
            <div
              className="col-lg-8 text-center"
            >
              <h2
                className="text-white mt-0"
              >
                Pizzas demo App by Fabian Calsina
              </h2>
              <hr
                className="divider light my-4"
              />
              <p
                className="text-white-50 mb-4"
              >
                <strong>
                  Back end:
                </strong>
                .Net webapi2, Entity Framework usingMSSQLLocalDB
              </p>
              <p
                className="text-white-50 mb-4"
              >
                <strong>
                  Front end:
                </strong>
                React, redux, redux-thunk, bootstrap 4.0
              </p>
            </div>
          </div>
        </div>
      </section>
      <section
        className="page-section"
      >
        <div
          className="container"
        >
          <h2
            className="text-center mt-0"
          >
            At Your Service
          </h2>
          <div
            className="text-center"
          >
            <img
              src="fgg.JPG"
              width="300"
            />
          </div>
          <hr
            className="divider my-4"
          />
          <div
            className="row"
          >
            <div
              className="col-lg-3 col-md-6 text-center"
            >
              <div
                className="mt-5"
              >
                <i
                  className="fas fa-4x fa-gem text-primary mb-4"
                />
                <h3
                  className="h4 mb-2"
                >
                  Email
                </h3>
                <p
                  className="text-muted mb-0"
                >
                  fabian.calsina@gmail.com
                </p>
              </div>
            </div>
            <div
              className="col-lg-3 col-md-6 text-center"
            >
              <div
                className="mt-5"
              >
                <i
                  className="fas fa-4x fa-laptop-code text-primary mb-4"
                />
                <h3
                  className="h4 mb-2"
                >
                  Cellphone
                </h3>
                <p
                  className="text-muted mb-0"
                >
                  591 70534341
                </p>
              </div>
            </div>
            <div
              className="col-lg-3 col-md-6 text-center"
            >
              <div
                className="mt-5"
              >
                <i
                  className="fas fa-4x fa-globe text-primary mb-4"
                />
                <h3
                  className="h4 mb-2"
                >
                  Github Profile
                </h3>
                <a
                  href="https://github.com/Fabho"
                  target="_blank"
                >
                  Fabho profile
                </a>
              </div>
            </div>
            <div
              className="col-lg-3 col-md-6 text-center"
            >
              <div
                className="mt-5"
              >
                <i
                  className="fas fa-4x fa-heart text-primary mb-4"
                />
                <h3
                  className="h4 mb-2"
                >
                  Made with Love
                </h3>
                <p
                  className="text-muted mb-0"
                >
                  2020
                </p>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>,
]
`;
